File: 프로그래머스/Level1/최대공약수와 최소공배수.py
# https://programmers.co.kr/learn/courses/30/lessons/12940

def solution(n, m):
    min_num = min(n, m)
    gcd = -1
    for i in range(min_num, 0,-1):
        if n % i == 0 and m % i == 0:
            gcd = i
            break
    return [gcd, gcd * (n // gcd) * (m // gcd)]

File: 프로그래머스/Level1/최대공약수와 최소공배수.js
// https://programmers.co.kr/learn/courses/30/lessons/12940

const solution = (n, m) => {
	const minValue = Math.min(n, m);
	let gcd = 1;
	for (let i = 2; i <= minValue; i++) {
		if (n % i === 0 && m % i === 0) gcd = i;
	}
	return [gcd, (n / gcd) * m];
};
